packages/web3-eth-contract/src/types.ts

```diff
@@ -34,40 +42,40 @@ export interface ContractOptions {
 	readonly from?: Address;
 	readonly data?: Bytes;
 	jsonInterface: ContractAbi;
-	address?: Address | null;
+	address?: Address;
 }
 
 export interface ContractInitOptions {
-	readonly gas: Uint | null;
-	readonly gasPrice: Uint | null;
+	readonly gas?: Uint | null;
+	readonly gasPrice?: Uint | null;
 	readonly from?: Address;
 	readonly data?: Bytes;
-	readonly gasLimit: Uint;
+	readonly gasLimit?: Uint;
 	readonly provider: SupportedProviders<EthExecutionAPI> | string;
 }
 
 export type TransactionReceipt = ReceiptInfo;
 
-export interface NonPayableTx {
+export interface NonPayableCallOptions {
 	nonce?: Numbers;
 	chainId?: Numbers;
 	from?: Address;
 	to?: Address;
 	data?: HexString;
-	gas?: Numbers;
+	gas?: string;
 	maxPriorityFeePerGas?: Numbers;
 	maxFeePerGas?: Numbers;
-	gasPrice?: Numbers;
+	gasPrice?: string;
 }
 
-export interface PayableTx extends NonPayableTx {
-	value?: Numbers;
+export interface PayableCallOptions extends NonPayableCallOptions {
+	value?: string;
```

Comment:
Why not `Numbers`?

Reply:
Same as https://github.com/ChainSafe/web3.js/pull/4678#discussion_r782115733